addons/test_mail/tests/test_message_post.py

```diff
@@ -1477,46 +1481,103 @@ def setUpClass(cls):
+        self.assertFalse(self.test_records[1].message_follower_ids)
+
+        self.assertEqual(self.user_employee.lang, 'en_US')
+
     @users('employee')
     @mute_logger('odoo.addons.mail.models.mail_mail')
     def test_composer_lang_template_comment(self):
+        """ When posting in comment mode, content is rendered using the lang
+        field of template. Notification layout lang is partly coming from
+        template, partly from environment (to be improved). """
         test_record = self.test_records[0].with_user(self.env.user)
         test_template = self.test_template.with_user(self.env.user)
 
-        with self.mock_mail_gateway():
-            test_record.message_post_with_source(
-                test_template,
-                email_layout_xmlid='mail.test_layout',
-                message_type='comment',
-                subtype_id=self.env.ref('mail.mt_comment').id,
-            )
-
-        record0_customer = self.env['res.partner'].search([('email_normalized', '=', 'test.record.1@test.customer.com')], limit=1)
-        self.assertTrue(record0_customer, 'Template usage should have created a contact based on record email')
-
-        customer_email = self._find_sent_mail_wemail(record0_customer.email_formatted)
-        self.assertTrue(customer_email)
-        body = customer_email['body']
-        # check content
-        self.assertIn(f'SpanishBody for {test_record.name}', body,
-                      'Body based on template should be translated')
-        # check subject
-        self.assertEqual(f'SpanishSubject for {test_record.name}', customer_email['subject'],
-                         'Subject based on template should be translated')
-        # check notification layout content
-        self.assertIn('Spanish Layout para', body, 'Layout content should be translated')
-        self.assertNotIn('English Layout for', body)
-        self.assertIn('Spanish Layout para Spanish Model Description', body, 'Model name should be translated')
-        # check notification layout strings
-        self.assertIn('View Lang Chatter Model', body,
-                      'Fixme: "View document" should be translated')
-        # self.assertIn('SpanishView Spanish Model Description', body,
-        #               '"View document" should be translated')
-        # self.assertNotIn(f'View {test_record._description}', body,
-        #                  '"View document" should be translated')
-        # self.assertIn('SpanishButtonTitle', body,
-        #               'Groups-based action names should be translated')
-        self.assertIn('NotificationButtonTitle', body, 'Fixme: Groups-based action names should be translated')
+        for partner, exp_content_lang, exp_notif_lang in zip(
+            (self.env['res.partner'], self.partner_1, self.partner_2),
+            ('es_ES', 'en_US', 'es_ES'),  # content: coming from template definition
+            ('en_US', 'en_US', 'es_ES'),  # layout: coming from customer
```

Comment:
Actually not sure why those are here.

It's not given to the "message_post_with_source" call so it's the values of something on the template / customer?
Should we assert that against the related fields then? (Note that it seems done for the customer but not for the template definition)
Or can't we use the related fields instead of giving this as test input?
There is only 1 template yet the values are changing multiple times?
It's a bit confusing.